src/components/search/Filter/Filter.tsx

```diff
@@ -145,13 +140,26 @@ function Filter({
   }
 
   const onApply = () => {
-    // Only toggle new facets and keep the current ones applied
-    const facetsToAdd = selectedFacets
-      .map((facet) => !searchState.selectedFacets.includes(facet) && facet)
-      .concat(facetsToRemove)
-      .filter((facet) => typeof facet !== 'boolean') as IStoreSelectedFacet[]
+    // Only toggle new facets added and old facets removed and keep the current ones applied
+    const facetsToAdd = selectedFacets.filter((facet) => {
+      const isSameFacet = (f: IStoreSelectedFacet) =>
+        f.value === facet.value && f.key === facet.key
+
+      const searchStateHasFacet = searchState.selectedFacets.some(isSameFacet)
+
+      return !searchStateHasFacet
+    })
+
+    const facetsToRemove = searchState.selectedFacets.filter((facet) => {
+      const isSameFacet = (f: IStoreSelectedFacet) =>
+        f.value === facet.value && f.key === facet.key
+
+      const isFacetSelected = selectedFacets.some(isSameFacet)
+
+      return !isFacetSelected
+    })
```

Comment:
What do you think of extracting a function here? It's the same logic, but for different instances of selected facets.

Reply:
That's true! I'm going to implement it. Thanks!